### PR TITLE
WIP Add certbot

### PR DESCRIPTION
**Pull request by bear-redhat**

### CODE REVIEW

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-service-main-e2e-ols-cluster

**Comment by bear-redhat** [comment]
/pj-rehearse ci/rehearse/openshift/lightspeed-service/main/e2e-ols-cluster

**Comment by bear-redhat** [comment]
/pj-rehearse ci/rehearse/openshift/lightspeed-service/main/e2e-ols-cluster

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-service-main-e2e-ols-cluster

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-service-main-e2e-ols-cluster

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-service-main-e2e-ols-cluster

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-service-main-e2e-ols-cluster

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-service-main-e2e-ols-cluster

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-service-main-e2e-ols-cluster

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-service-main-e2e-ols-cluster

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-service-main-e2e-ols-cluster

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-service-main-e2e-ols-cluster

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-service-main-e2e-ols-cluster

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-service-main-e2e-ols-cluster

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-service-main-e2e-ols-cluster

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-service-main-e2e-ols-cluster

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-service-main-e2e-ols-cluster

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-service-main-e2e-ols-cluster

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-service-main-e2e-ols-cluster

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-service-main-e2e-ols-cluster

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-service-main-e2e-ols-cluster

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-service-main-e2e-ols-cluster

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-service-main-e2e-ols-cluster

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-service-main-e2e-ols-cluster

File: ci-operator/step-registry/hypershift/hostedcluster/certbot/hypershift-hostedcluster-certbot-commands.sh [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+if [[ $? -ne 0 ]]; then
+  >&2 echo "ERROR: Failed to obtain a wildcard certificate for ${CLUSTER_DOMAIN}"
+  exit 1
+fi
+
+echo "Successfully obtained a wildcard certificate for ${CLUSTER_DOMAIN}"
+
+echo "Installing the certificate..."
+/cli/oc -n clusters create secret tls ${CLUSTER_NAME}-cert \
```
Comment (danilo-gemoli) [inline ci-operator/step-registry/hypershift/hostedcluster/certbot/hypershift-hostedcluster-certbot-commands.sh:82]: Ok I have realized we expose `$CLI_DIR`, therefore `"$CLI_DIR/oc"` might be better.

File: ci-operator/step-registry/hypershift/hostedcluster/certbot/hypershift-hostedcluster-certbot-commands.sh [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    --cert=${CONFIG_DIR}/live/${CLUSTER_DOMAIN}/fullchain.pem \
+    --key=${CONFIG_DIR}/live/${CLUSTER_DOMAIN}/privkey.pem
+
+#configure hostedcluster object with cluster name to set
+# hostedcluster.spec.configuration.apiServer.servigCerts
+echo "Setting the servingCerts for API server"
+cat <<EOF | /cli/oc patch -n clusters -f - --type merge
```
Comment (danilo-gemoli) [inline ci-operator/step-registry/hypershift/hostedcluster/certbot/hypershift-hostedcluster-certbot-commands.sh:89]: Same as above.

**Review by danilo-gemoli — CHANGES_REQUESTED** [review]

File: ci-operator/step-registry/hypershift/hostedcluster/certbot/hypershift-hostedcluster-certbot-commands.sh (hunk 1 above)
Comment (bear-redhat) [reply ci-operator/step-registry/hypershift/hostedcluster/certbot/hypershift-hostedcluster-certbot-commands.sh:83]: Thanks

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-service-main-e2e-ols-cluster

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-service-main-e2e-ols-cluster

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-openshift-lightspeed-service-main-e2e-ols-cluster

**Pull request closed without merging**